src/components/bbox/Bbox.tsx

```diff
@@ -118,45 +119,61 @@ const BboxDiv = styled.div`
+    onClick,
+  } = props;
+
   const [left, bottom, width, height, top] = useMemo(() => {
     return [
-      (parseFloat(props.bbox.location[0] as string) * props.scale) + 'px',
-      (props.bbox.location[3] === 'bottom'
-        ? '0'
-        : (parseFloat(props.bbox.location[1] as string) * props.scale) + 'px'),
-      (parseFloat(props.bbox.location[2] as string) * props.scale) + 'px',
-      (props.bbox.location[3] === 'top'
-        ? 'auto'
-        : (parseFloat(props.bbox.location[3] as string) * props.scale) + 'px'),
-      props.bbox.location[3] === 'top'
-        ? '0' : props.bbox.location[3] === 'bottom'
-        ? `calc(100% - ${parseFloat(props.bbox.location[1] as string) * props.scale}px)`
-        : 'auto',
+      (parseFloat(bbox.location[0] as string) * scale) + 'px',
+      (bbox.location[3] === 'bottom'
+          ? '0'
+          : (parseFloat(bbox.location[1] as string) * scale) + 'px'),
+      (parseFloat(bbox.location[2] as string) * scale) + 'px',
+      (bbox.location[3] === 'top'
+          ? 'auto'
+          : (parseFloat(bbox.location[3] as string) * scale) + 'px'),
+      bbox.location[3] === 'top'
+          ? '0' : bbox.location[3] === 'bottom'
+              ? `calc(100% - ${parseFloat(bbox.location[1] as string) * scale}px)`
+              : 'auto',
     ]
-  }, [props.bbox.location, props.scale]);
+  }, [bbox.location, scale]);
 
-  const isSelected = useMemo(() => props.selected ? ' pdf-bbox_selected' : '', [props.selected]);
-  const isRelated = useMemo(() => props.related ? ' pdf-bbox_related' : '', [props.related]);
-  const isDisabled = useMemo(() => props.disabled ? ' pdf-bbox_disabled' : '', [props.disabled]);
-  const isStructured = useMemo(() => props.structured ? ' pdf-bbox_structured' : '', [props.structured]);
-  const isStructuredSelected = useMemo(() => props.structured && props.selected ? ' pdf-bbox_structured_selected' : '', [props.structured, props.selected]);
+  const isSelected = useMemo(() => selected ? ' pdf-bbox_selected' : '', [selected]);
+  const isRelated = useMemo(() => related ? ' pdf-bbox_related' : '', [related]);
+  const isDisabled = useMemo(() => disabled ? ' pdf-bbox_disabled' : '', [disabled]);
+  const isStructured = useMemo(() => structured ? ' pdf-bbox_structured' : '', [structured]);
+  const isStructuredSelected = useMemo(() => structured && selected ? ' pdf-bbox_structured_selected' : '', [structured, selected]);
   const isStructuredSelectedMultiple = useMemo(() => {
-    if (props.structured && props.selected && props.selectionMode === TreeBboxSelectionMode.SELECTED_WITH_KIDS) return ' pdf-bbox_structured_selected_multiple';
+    if (structured && selected && selectionMode === TreeBboxSelectionMode.SELECTED_WITH_KIDS) return ' pdf-bbox_structured_selected_multiple';
     else return '';
-  }, [props.structured, props.selected, props.selectionMode]);
+  }, [structured, selected, selectionMode]);
+
+  if (renderBbox) {
+    return renderBbox(props);
```

Comment:
We better pass all transformed location info too (the `const [left, bottom, width, height, top]` variabled defined above) to make sure bboxes are rendered at the correct place and no extra calculations needed.